app/src/main/java/org/schabi/newpipe/util/SparseItemUtil.java

```diff
@@ -0,0 +1,126 @@
+package org.schabi.newpipe.util;
+
+import static org.schabi.newpipe.extractor.stream.StreamType.AUDIO_LIVE_STREAM;
+import static org.schabi.newpipe.extractor.stream.StreamType.LIVE_STREAM;
+import static org.schabi.newpipe.extractor.utils.Utils.isNullOrEmpty;
+
+import android.content.Context;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import org.schabi.newpipe.NewPipeDatabase;
+import org.schabi.newpipe.R;
+import org.schabi.newpipe.database.stream.model.StreamEntity;
+import org.schabi.newpipe.error.ErrorInfo;
+import org.schabi.newpipe.error.ErrorUtil;
+import org.schabi.newpipe.error.UserAction;
+import org.schabi.newpipe.extractor.stream.StreamInfo;
+import org.schabi.newpipe.extractor.stream.StreamInfoItem;
+import org.schabi.newpipe.player.playqueue.SinglePlayQueue;
+
+import java.util.function.Consumer;
+
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers;
+import io.reactivex.rxjava3.core.Completable;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+
+/**
+ * Utility class for fetching additional data for stream items when needed.
+ */
+public final class SparseItemUtil {
+    private SparseItemUtil() {
+    }
+
+    /**
+     * Use this to certainly obtain an single play queue with all of the data filled in when the
+     * stream info item you are handling might be sparse, e.g. because it was fetched via a {@link
+     * org.schabi.newpipe.extractor.feed.FeedExtractor}. FeedExtractors provide a fast and
+     * lightweight method to fetch info, but the info might be incomplete (see
+     * {@link org.schabi.newpipe.local.feed.service.FeedLoadService} for more details).
+     *
+     * @param context the Android context
+     * @param item the item which is checked and eventually loaded completely
+     * @param callback the callback to call with the single play queue built from the original item
+     *                 if all info was available, otherwise from the fetched {@link
+     *                 org.schabi.newpipe.extractor.stream.StreamInfo}
+     */
```

Comment:
Indent the parameters description so they are all aligned on the same column (like that's the case in most of other places).